### ISSUE TITLE
Swiftlint Required version is triggered even though version is higher

### ISSUE THREAD

**Issue by christiankm** [realm/SwiftLint#2702]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

We're using swiftlint_version: 0.29.0 in our configuration file, but even though im running a higher version of swiftlint, I get an error. I realize that this parameter defines the exact version, but I would expect as a user, that it defines a minimum version.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Found a configuration for 'force_unwrapping' rule, but it is not enabled on 'opt_in_rules'.
Currently running SwiftLint 0.31.0 but configuration specified version 0.29.0.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?: 0.31.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?: home-brew
* Paste your configuration file:

```yml
swiftlint_version: 0.29.0

...
```

**Comment by marcelofabri** [comment]
This is the expected behavior. Check #2592 for a similar issue.
